src/main/kotlin/minesweeper/domain/tile/strategy/RandomGenerateStrategy.kt

```diff
@@ -0,0 +1,19 @@
+package minesweeper.domain.tile.strategy
+
+import minesweeper.domain.MineCount
+import minesweeper.domain.tile.pos.Coordinate
+import minesweeper.domain.tile.pos.Position
+import minesweeper.domain.tile.state.Tile
+
+class RandomGenerateStrategy(override val width: Position, override val height: Position, override val mineCount: MineCount) : GenerateStrategy {
+    override fun generate(): List<Tile> {
+        val tiles = mutableListOf<Tile>()
+        var mountCount = mineCount.value
+
+        for (x in 0..width.value) for (y in 0..height.value) {
+            val coordinate = Coordinate(Position(x), Position(y))
+            tiles.add(mountCount--.let { if (it > 0) Tile.of(coordinate, true) else Tile.of(coordinate, false) })
```

Comment:
https://github.com/next-step/kotlin-minesweeper/pull/195#discussion_r1055033093 이 코멘트에 이어서...
다음처럼 Tile에 정말 메세지만 던지는 것을 의도했습니다! 구현은 최대한 객체 내에 감춰두고 외부에서는 메세지만 던져보세요!
```suggestion
            tiles.add(Tile.of(coordinate, mountCount--))
```

Reply:
전달해 주신 코멘트 꼭 기억해 두겠습니다! 제가 처음에 너무 복잡하게 생각했네요... 조언해 주신 방법대로 해보니 코드가 훨씬 깔끔해 졌습니다 😊

> 구현은 최대한 객체 내에 감춰두고 외부에서는 메세지만 던져보세요!

- https://github.com/next-step/kotlin-minesweeper/commit/72b041dd0ae8faa26ded77aea40dbd2de8d33d58